networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerSelectionStrategy.java

```diff
@@ -180,12 +231,86 @@ public List<Peer> selectPeersToDisconnect(
     // for disconnection based on their score
     randomlySelectedPeersBeingDropped.forEach(
         peer -> peerPools.addPeerToPool(peer.getId(), SCORE_BASED));
-    return Stream.concat(
-            randomlySelectedPeersBeingDropped.stream(),
-            peersBySource.getOrDefault(SCORE_BASED, emptyList()).stream())
-        .sorted(Comparator.comparing(peerScorer::scoreExistingPeer))
-        .limit(peersToDrop)
-        .toList();
+
+    if (peersToDrop > falseAdvertisingPeers.size()) {
+      return Stream.concat(
+              falseAdvertisingPeers.stream(),
+              Stream.concat(
+                      randomlySelectedPeersBeingDropped.stream(),
+                      peersBySource.getOrDefault(SCORE_BASED, emptyList()).stream())
+                  .sorted(Comparator.comparing(peerScorer::scoreExistingPeer))
+                  .limit(peersToDrop - falseAdvertisingPeers.size()))
+          .toList();
+    } else {
+      return falseAdvertisingPeers;
+    }
+  }
+
+  private Optional<Peer> verifyPeerAdvertisement(
+      final Peer peer,
+      final Map<NodeId, List<String>> topicsBySubscriber,
+      final Map<UInt256, DiscoveryPeer> discoveryPeerMap,
+      final PeerSubnetSubscriptions peerSubnetSubscriptions) {
+    if (activeEth2P2PNetworkReference.get() == null) {
+      return Optional.empty();
+    }
+    final ActiveEth2P2PNetwork activeEth2P2PNetwork = activeEth2P2PNetworkReference.get();
+
+    final List<String> actualPeerTopics = topicsBySubscriber.get(peer.getId());
+    if (actualPeerTopics.isEmpty()) {
+      return Optional.empty();
+    }
+
+    final Optional<Eth2Peer> eth2Peer = activeEth2P2PNetwork.getPeer(peer.getId());
+    if (eth2Peer.isEmpty()) {
+      return Optional.empty();
+    }
+
+    final Optional<UInt256> discoveryNodeId = eth2Peer.get().getDiscoveryNodeId();
+    if (discoveryNodeId.isEmpty()) {
+      return Optional.empty();
+    }
+
+    if (!discoveryPeerMap.containsKey(discoveryNodeId.get())) {
+      return Optional.empty();
+    }
+
+    final DiscoveryPeer discoveryPeer = discoveryPeerMap.get(discoveryNodeId.get());
+
+    // Check for sync committee subnets
+    final SszBitvector advertisedSyncCommitteeSubnets = discoveryPeer.getSyncCommitteeSubnets();
+    final SszBitvector syncCommitteeSubnets =
+        peerSubnetSubscriptions.getAttestationSubnetSubscriptions(peer.getId());
+    if (!syncCommitteeSubnets.isSuperSetOf(advertisedSyncCommitteeSubnets)) {
+      reputationManager.adjustReputation(peer.getAddress(), ReputationAdjustment.LARGE_PENALTY);
```

Comment:
Maybe we could add a debug entry here saying we are downscoring peer X (not sure if we log this somewhere else). This will make it a bit easier to see this mechanism in action.